### PR TITLE
Multiobs initialization and moments conversion

### PR DESCRIPTION
**Pull request by b-remy**
Hi @pmelchior ,

I'm opening this PR to make the code review for #73  easier. I added on my side the code for converting moments from one WCS to another. I also added a few tests targeting rescaling, rotation, flip and arbitrary WCS conversion, including the case where the observations are mono or multichannels. You can also take a look at [this notebook](https://colab.research.google.com/drive/1BspxcnCr2wmIn_HSuYRve2mttovhRU19?usp=sharing) which illustrates these tests :-)

### CODE REVIEW

File: scarlet2/measure.py [hunk 1]
```diff
@@ -124,71 +126,215 @@ def moments(component, N=2, center=None, weight=None):
...
+                    center = jnp.asarray(center)
+                    # centroid wrt given center
+                    self[1, 0] -= center[0] * self[0, 0]
+                    self[0, 1] -= center[1] * self[0, 0]
+                if model.ndim == 3 and center[0].ndim == 1:
+                    center = center[0][:, None, None], center[1][:, None, None]
+                grid_y = grid_y - center[0]
+                grid_x = grid_x - center[1]
+    
+    def higher_order_moments(self):
```
Comment (pmelchior) [inline scarlet2/measure.py:188]: What is this function for? From the facto that it goes from moment N to 2N, I'm guessing to get the variances...

However, the constructor can measure any moment you want, but this code requires the model and the weights to be saved, which I am not a fan of because there's ample change to modify one without the others updating.

File: tests/test_quickstart.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ def test_quickstart():
         for center in centers:
             center = jnp.array(center)
             try:
-                spectrum, morph = init.adaptive_morphology(obs, center, min_corr=0.99)
+                spectrum, morph = init.make_bbox(obs, center, min_corr=0.99)
```
Comment (pmelchior) [inline tests/test_quickstart.py:32]: This must be incorrect. That method only returns a box, but we need `spectrum, morph = init.adaptive_morphology(obs, center, min_corr=0.99)`. Please change back.

**Comment by pmelchior** [comment]
Overall, this looks very good. Happy to standardize moments. This will be useful for good inits, provided we there's no garbage in the image, like neighboring galaxies ;-)

File: scarlet2/measure.py (hunk 1 above)
Comment (b-remy) [reply scarlet2/measure.py:188]: In order to update moments from 0 to N, we also need to use moments from N+1 to 2N (see [Teague (1980), eq. 36](https://opg.optica.org/josa/fulltext.cfm?uri=josa-70-8-920&id=57703)), so I proposed to compute these additional moments in another function to update the first N moments stored the class. I am not a fan either that it requires to have access to the model. The problem is that once we updated the first N moments, the moments up to 2N are outdated since still correspond to the untransformed original image, so cannot be used for another transformation... 

Unless we keep track of the transformations in the class, and if the object is affected by another transformation we know the corresponding higher moments of the original image. What do you think?

Comment (pmelchior) [reply scarlet2/measure.py:188]: I don't see what you mean. Teague's eq. 36 clearly shows that you don't need higher moments. The paper even clarifies this point in the text below.

File: scarlet2/initialization.py [hunk 3]
```diff
@@ -23,9 +25,9 @@ def _get_edge_pixels(img, box):
     return jnp.concatenate(edge, axis=1)
 
 
-def adaptive_morphology(obs, center, min_size=11, delta_size=3, min_snr=20, min_corr=0.99, return_array=False):
+def make_bbox(obs, center_pix, min_size=11, delta_size=3, min_snr=20, min_corr=0.99):
     """
-    Create image of a Gaussian from the centered 2nd moments of the observation.
+    Make a bounding box for source at center.
```
Comment (b-remy) [inline scarlet2/initialization.py:30]: This function, which used to return the adaptive morphology, now only returns the box. I agree we should revert the change, but I wanted to double check with you if this was intended. I'm happy to make the change back.

File: tests/test_quickstart.py (hunk 2 above)
Comment (b-remy) [reply tests/test_quickstart.py:32]: Right, I agree! Please check the related comment bellow

Comment (pmelchior) [reply tests/test_quickstart.py:32]: OK, we're both wrong. The drop-in replacement for this method is `init.from_gaussian_moments`. Please change that

File: scarlet2/initialization.py (hunk 3 above)
Comment (pmelchior) [reply scarlet2/initialization.py:30]: Yes, that is fine. I made the code more modular, so this function was broken out. No changes needed here.

File: scarlet2/measure.py (hunk 1 above)
Comment (pmelchior) [reply scarlet2/measure.py:188]: This still is an open issue. I would much prefer the `Moment` instances not holding on to the data they were computed with.

File: tests/test_quickstart.py (hunk 2 above)
Comment (b-remy) [reply tests/test_quickstart.py]: Right, this is fixed now :-)

File: scarlet2/measure.py (hunk 1 above)
Comment (b-remy) [reply scarlet2/measure.py]: I got confused due to the j+k-r-s index, which I thought could go up to 4 if j and k are both equal to 2. But the sum is never above 2 for 2nd order moments (j=m, k=n-m)

Comment (b-remy) [reply scarlet2/measure.py]: This should be fixed now :-)

**Review by pmelchior — APPROVED** [review]
all good. go ahead and merge
